### PR TITLE
4-get-reviews, added jest-sorted

### PR DESCRIPTION
**Pull request by reawilliams20**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-const {
-getCategories
-} = require("./controllers/controllers.categories")
-const {handle404Routes} = require("./controllers/controllers.errors")
+const {getCategories} = require("./controllers/controllers.categories")
+const { getReviews } = require("./controllers/controllers.reviews");
+const {handle404Routes} = require("./controllers/controllers.errors");
 
 const app = express();
 app.use(express.json());
```
Comment (lewispricey) [inline app.js]: Please could you remove express.json as it's not required at this stage

**Comment by reawilliams20** [comment]
Sorry, this is not finished yet. I will upload another commit later

File: __tests__/app.test.js [hunk 2]
```diff
@@ -46,4 +46,41 @@ describe('/api/categories', () => {
...
+
+describe('/api/reviews', () => {
+    test('200: OK, returns with reviews', () => {
+        return request(app)
+        .get('/api/reviews')
+        .expect(200)
+        .then(({body : {reviews}}) => {
+            expect(reviews).toBeInstanceOf(Array);
+            expect(reviews).toHaveLength(13);
+            expect(reviews).toBeSortedBy('created_at', { descending: true });
```
Comment (lewispricey) [inline __tests__/app.test.js:51]: Great use of jest-sorted to check the values are in the required order

File: __tests__/app.test.js [hunk 3]
```diff
@@ -46,4 +46,41 @@ describe('/api/categories', () => {
...
+                        category: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number),
+                        comment_count: expect.any(String)
+                    })
+                )
+            })
+        })
+    });
+    test('404: route not found', () => {
```
Comment (lewispricey) [inline __tests__/app.test.js]: This is a good test, I'd recommend testing for any invalid routes in a seperate test block as the 404 handling for an invalid route isn't directly related to the endpoints functionality

File: app.js [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
 
 app.get('/api/categories', getCategories);
+app.get('/api/reviews', getReviews);
```
Comment (lewispricey) [inline app.js:10]: Great naming convention for your controller functions!

File: controllers/controllers.reviews.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const {selectReviews} = require("../models/models.reviews");
+
+exports.getReviews = (req, res, next) => {
+    selectReviews()
+    .then((reviews) => {
+        res.status(200).send({reviews})
+    })
+    .catch((err) => {
```
Comment (lewispricey) [inline controllers/controllers.reviews.js:8]: I like that you've added a catch block here incase of any database connectivity issues

File: __tests__/app.test.js [hunk 6]
```diff
@@ -46,4 +46,41 @@ describe('/api/categories', () => {
...
+        return request(app)
+        .get('/api/reviews')
+        .expect(200)
+        .then(({body : {reviews}}) => {
+            expect(reviews).toBeInstanceOf(Array);
+            expect(reviews).toHaveLength(13);
+            expect(reviews).toBeSortedBy('created_at', { descending: true });
+            reviews.forEach((review) => {
+                expect(review).toEqual(
+                    expect.objectContaining({
```
Comment (lewispricey) [inline __tests__/app.test.js:54]: Awesome property tests, really thorough.

**Review by lewispricey — APPROVED** [review]
Overall great work Andrea 👍 Please remove the express.json and then feel free to merge!

No need to send though another PR before merging 🚀

**Pull request merged**